File: src/main/kotlin/net/shadowfacts/craftingslabs/property/PropertyDummy.java
package net.shadowfacts.craftingslabs.property;

import com.google.common.base.Optional;
import com.google.common.collect.ImmutableList;
import net.minecraft.block.properties.PropertyHelper;
import org.jetbrains.annotations.NotNull;

import java.util.Collection;

/**
 * @author shadowfacts
 */
public class PropertyDummy extends PropertyHelper<PropertyDummy.Dummy> {

	public PropertyDummy(String name) {
		super(name, Dummy.class);
	}

	@Override
	public Collection<Dummy> getAllowedValues() {
		return ImmutableList.of(Dummy.INSTANCE);
	}

	@Override
	public Optional<Dummy> parseValue(String value) {
		return Optional.of(Dummy.INSTANCE);
	}

	@Override
	public String getName(Dummy value) {
		return "dummy";
	}

	public static class Dummy implements Comparable<Dummy> {
		public static final Dummy INSTANCE = new Dummy();
		private Dummy() {}

		@Override
		public int compareTo(@NotNull Dummy o) {
			return 0;
		}
	}

}

File: src/main/kotlin/net/shadowfacts/craftingslabs/property/PropertyDummy.kt
package net.shadowfacts.craftingslabs.property

import com.google.common.base.Optional
import net.minecraft.block.properties.PropertyHelper

/**
 * @author shadowfacts
 */
class PropertyDummy(name: String): PropertyHelper<PropertyDummy.Dummy>(name, Dummy::class.java) {

	object Dummy: Comparable<Dummy> {

		override fun compareTo(other: Dummy): Int {
			return 0
		}
	}

	override fun getName(value: Dummy?): String {
		return ""
	}

	override fun getAllowedValues(): MutableCollection<Dummy> {
		return mutableSetOf(Dummy)
	}

	override fun parseValue(value: String?): Optional<Dummy> {
		return Optional.of(Dummy)
	}

}
